Source/Core/Core/WiiUtils.cpp

```diff
@@ -109,14 +110,53 @@ static bool InstallWAD(IOS::HLE::Kernel& ios, const DiscIO::WiiWAD& wad)
   return true;
 }
 
+bool InstallWAD(IOS::HLE::Kernel& ios, const DiscIO::WiiWAD& wad, InstallType install_type)
+{
+  if (!wad.GetTMD().IsValid())
+    return false;
+
+  // Skip the install if the WAD is already installed.
+  const auto installed_contents = ios.GetES()->GetStoredContentsFromTMD(wad.GetTMD());
+  if (wad.GetTMD().GetContents() == installed_contents)
+    return true;
+
+  // If a different version is currently installed, warn the user to make sure
+  // they don't overwrite the current version by mistake.
+  const u64 title_id = wad.GetTMD().GetTitleId();
+  const bool has_another_version =
+      ios.GetES()->FindInstalledTMD(title_id).IsValid() &&
+      std::any_of(installed_contents.begin(), installed_contents.end(),
+                  [](const IOS::ES::Content& content) { return !content.IsShared(); });
+  if (has_another_version &&
+      !AskYesNoT("A different version of this title is already installed on the NAND. "
```

Comment:
Not sure if this makes sense for the average WAD, but...does adding the two version numbers to the message add any value for the end user?

Reply:
I think it would (to let the user know if it's a downgrade/upgrade), so I've added version numbers to the message.